File: src/main/java/eterea/migration/api/rest/model/Product.java
package eterea.migration.api.rest.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(uniqueConstraints = {
    @UniqueConstraint(columnNames = {"orderNumberId", "lineId"})
})
public class Product extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;
    private Long orderNumberId;
    private String sku;
    private int lineId;
    private String name;
    private int qty;
    private BigDecimal itemPrice;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime bookingStart;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime bookingEnd;
    private int bookingDuration;
    private int bookingPersons;
    private String personTypes;
    private String serviciosAdicionales;
    private String puntoDeEncuentro;
    private String encuentroHotel;
}

File: src/main/kotlin/eterea/migration/api/rest/kotlin/model/Product.kt
package eterea.migration.api.rest.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import eterea.migration.api.rest.model.Auditable
import jakarta.persistence.*
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["orderNumberId", "lineId"])])
data class Product(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var productId: Long? = null,
    var orderNumberId: Long? = null,
    var sku: String? = "",
    var lineId: Int = 0,
    var name: String = "",
    var qty: Int = 0,
    var itemPrice: BigDecimal = BigDecimal.ZERO,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var bookingStart: OffsetDateTime? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var bookingEnd: OffsetDateTime? = null,
    var bookingDuration: Int = 0,
    var bookingPersons: Int = 0,
    var personTypes: String = "",
    var serviciosAdicionales: String = "",
    var puntoDeEncuentro: String = "",
    var encuentroHotel: String = "",

    @ManyToOne
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    var orderNote: OrderNote? = null

) : Auditable()
